src/io/parser.cpp

```diff
@@ -70,12 +70,19 @@ enum DataType {
 };
 
 Parser* Parser::CreateParser(const char* filename, bool has_header, int num_features, int label_idx) {
-  std::ifstream tmp_file;
-  tmp_file.open(filename);
-  if (!tmp_file.is_open()) {
-    Log::Fatal("Data file %s doesn't exist'", filename);
+  auto reader = VirtualFileReader::Make(filename);
+  if (!reader->Init()) {
+    Log::Fatal("Data file %s doesn't exist'", reader->filename_.c_str());
   }
   std::string line1, line2;
+  size_t buffer_size = 32*1024;
```

Comment:
I think this may is not enough for the long lines.

Reply:
I've updated this so it should work for any length line possible, and added a test on the python package to assert it's working.